rest/diagnostic_api.go

```diff
@@ -0,0 +1,69 @@
+//  Copyright 2013-Present Couchbase, Inc.
+//
+//  Use of this software is governed by the Business Source License included
+//  in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+//  in that file, in accordance with the Business Source License, use of this
+//  software will be governed by the Apache License, Version 2.0, included in
+//  the file licenses/APL2.txt.
+
+package rest
+
+import (
+	"fmt"
+
+	"github.com/couchbase/sync_gateway/auth"
+	"github.com/couchbase/sync_gateway/base"
+	channels "github.com/couchbase/sync_gateway/channels"
+	"github.com/gorilla/mux"
+)
+
+type allChannels struct {
+	Channels map[string]map[string]auth.GrantHistory `json:"all_channels,omitempty"`
```

Comment:
As discussed offline, I think that it probably makes sense to remove `updated_at` and so this struct become a new struct of

```
type channelHistory struct {
	Entries   []GrantHistorySequencePair `json:"entries"`    // Entry for a specific grant period
}
```

We don't always have the time the UpdatedAt was - for access grants, this value is zero.